src/vld.cpp

```diff
@@ -2925,6 +2921,11 @@ CaptureContext::~CaptureContext() {
         return;
 
     if ((m_tls->blockWithoutGuard) && (!IsExcludedModule())) {
+        CallStack* callstack = CallStack::Create();
+        callstack->getStackTrace(g_vld.m_maxTraceFrames, m_tls->context);
+
+        CriticalSectionLocker<> cs(g_heapMapLock);
```

Comment:
shouldnt this be before getStackTrace to avoid a race where the stacktrace is gotten, then modified by another thread, then modified here?

Reply:
well, the stacktrace should refer to the current thread, so no, another thread would not modify it.